### ISSUE TITLE
Relocate and refactor `woo-test-helper.php`

### ISSUE THREAD

**Issue by nielslange** [woothemes/woocommerce#46745]
> [!CAUTION]
> This issue is blocked by #46125 as it should build on top of it.

### Describe the solution you'd like

In #46125, @WunderBart was working on eliminating the side effects of our Playwright e2e tests. During a review, I noticed that while most helper plugins have been moved to `plugins/woocommerce-blocks/tests/e2e/plugins`, there's one more helper plugin `woo-test-helper.php`, which is currently located at `plugins/woocommerce-blocks/tests/mocks/woo-test-helper`.

In comparison to other plugins, this plugin needs to be activated when setting up the e2e environment. Certain plugin actions are triggered by calling certain URL parameters, e.g. `${ process.env.WORDPRESS_BASE_URL }/?setup_terms_and_privacy`. In https://github.com/woocommerce/woocommerce/pull/46125#discussion_r1565848437, @WunderBart suggested not to move `woo-test-helper.php` to `plugins/woocommerce-blocks/tests/e2e/plugins` as part of his PR, given that the PR is already enormous.

When working on this issue, we should carefully evaluate the various sections of the `woo-test-helper.php` helper plugin and potentially split it into smaller, more targeted helper plugins. For example, currently, the helper plugin sets up and tears down the terms and conditions page and the privacy policy page, as well as registering third part local pickup methods.
